contracts/TwabRewards.sol

```diff
@@ -0,0 +1,379 @@
+import "@pooltogether/owner-manager-contracts/contracts/Manageable.sol";
+
+import "./interfaces/ITwabRewards.sol";
+
+/**
+ * @title PoolTogether V4 TwabRewards
+ * @author PoolTogether Inc Team
+ * @notice Contract to distribute rewards to depositors in a pool.
+ */
+contract TwabRewards is ITwabRewards, Manageable {
+    using SafeERC20 for IERC20;
+
+    /* ============ Global Variables ============ */
+
+    /// @notice Settings of each promotion.
+    mapping(uint256 => Promotion) internal _promotions;
+
+    /// @notice Current promotion id.
+    uint256 internal _currentPromotionId;
+
+    /// @notice Keeps track of claimed rewards per user.
+    /// @dev _claimedRewards[promotionId][epochId][user] => uint256
+    mapping(uint256 => mapping(uint256 => mapping(address => uint256))) internal _claimedRewards;
+
+    /* ============ Events ============ */
+
+    /**
+        @notice Emitted when contract has been deployed.
+        @param owner Contract owner address
+   */
+    event Deployed(address owner);
+
+    /**
+        @notice Emmited when a promotion is created.
+        @param promotion Pomotion settings
+    */
+    event PromotionCreated(Promotion promotion);
+
+    /* ============ Constructor ============ */
+
+    /**
+        @notice Deploy TwabRewards contract.
+        @param _owner Contract owner address
+    */
+    constructor(address _owner) Ownable(_owner) {
+        emit Deployed(_owner);
+    }
+
+    /* ============ Modifiers ============ */
+
+    /// @dev Ensure that the caller is the creator of the currently active promotion.
+    modifier onlyPromotionCreator() {
+        require(msg.sender == _getCurrentPromotion().creator, "TwabRewards/only-promotion-creator");
+        _;
+    }
+
+    /* ============ External Functions ============ */
+
+    /// @inheritdoc ITwabRewards
+    function createPromotion(address _ticket, PromotionParameters calldata _promotionParameters)
+        external
+        override
+        returns (bool)
+    {
+        require(_isPromotionActive() == false, "TwabRewards/promotion-already-active");
+
+        _requireTicket(_ticket);
+
+        uint256 _nextPromotionId = _currentPromotionId + 1;
+        _currentPromotionId = _nextPromotionId;
+
+        address _token = _promotionParameters.token;
+        uint256 _tokensPerEpoch = _promotionParameters.tokensPerEpoch;
+        uint256 _numberOfEpochs = _promotionParameters.numberOfEpochs;
+
+        IERC20(_token).safeTransfer(address(this), _tokensPerEpoch * _numberOfEpochs);
+
+        Promotion memory _nextPromotion = Promotion(
+            uint32(_nextPromotionId),
+            false,
+            msg.sender,
+            _ticket,
+            _token,
+            _tokensPerEpoch,
+            _promotionParameters.startTimestamp,
+            _promotionParameters.epochDuration,
+            _numberOfEpochs
+        );
+
+        _promotions[_nextPromotionId] = _nextPromotion;
+
+        emit PromotionCreated(_nextPromotion);
```

Comment:
Just emit the promotion id!  The rest of the params are passed in so they're kind of redundant

Reply:
Done: https://github.com/pooltogether/v4-periphery/pull/8/commits/6f2b3064232970c0adb680a35671b97c89159672